mvp.feature

```diff
@@ -865,11 +866,12 @@ Feature: MVP features for NVA
   # Actions from Page : My Publications (Delete)
   @356
   Scenario: Creator deletes an item in My Publications list
-    Given that the user is logged in as Creator
-    And is on the My Publications page
+    Given the user is logged in as Creator
+    And has opened the page My Publications
     When they click Delete on an item
-    And they click Yes in the confirmation dialog
-    Then they see the publication is marked as Deleted
+    Then a comfirmation pop-up is opened
+    When the user selects Yes the publication is deleted (in the db)
```

Comment:
det skal egentlig ikke være noen `When`s etter en `Then`. 
Da må du isåfall skrive om featuren, og legge til en ny. F.eks. slik:
```
Scenario: Creator deletes an item in My Publications list
    Given Creator opens My Publications
    When they click Delete on an item
    And they see a confirmation pop-up is opened
    And they select Yes
    Then they see that the publication is deleted
```
og legge til denne:
```
Scenario: Creator does not delete an item in My Publications list
    Given Creator opens My Publications
    When they click Delete on an item
    And they see a confirmation pop-up is opened
    And they select No
    Then they see that the pop-up is closed
```

Reply:
I agree, the code smell is obvious…because we need to state "(in the DB)". There needs to be a verification of this, which needs to happen in a separate step. The Given->When->Then structure enforces focus on a single, logical function. There is a construct with "But", but  it is relatively(?) new and I have seen it once and I find it confusing and am unsure of its semantics in most cases.